tests/chainer_tests/datasets_tests/test_sub_dataset.py

```diff
@@ -0,0 +1,119 @@
+import unittest
+
+from chainer import datasets
+from chainer import testing
+
+
+class TestSubDataset(unittest.TestCase):
+
+    def test_sub_dataset(self):
+        original = [1, 2, 3, 4, 5]
+        subset = datasets.SubDataset(original, 1, 4)
+        self.assertEqual(len(subset), 3)
+        self.assertEqual(subset[0], 2)
+        self.assertEqual(subset[1], 3)
+        self.assertEqual(subset[2], 4)
+
+    def test_permuted_sub_dataset(self):
+        original = [1, 2, 3, 4, 5]
+        subset = datasets.SubDataset(original, 1, 4, [2, 0, 3, 1, 4])
+        self.assertEqual(len(subset), 3)
+        self.assertEqual(subset[0], 1)
+        self.assertEqual(subset[1], 4)
+        self.assertEqual(subset[2], 2)
+
+    def test_split_dataset(self):
+        original = [1, 2, 3, 4, 5]
+        subset1, subset2 = datasets.split_dataset(original, 2)
+        self.assertEqual(len(subset1), 2)
+        self.assertEqual(subset1[0], 1)
+        self.assertEqual(subset1[1], 2)
+        self.assertEqual(len(subset2), 3)
+        self.assertEqual(subset2[0], 3)
+        self.assertEqual(subset2[1], 4)
+        self.assertEqual(subset2[2], 5)
+
+    def test_permuted_split_dataset(self):
+        original = [1, 2, 3, 4, 5]
+        subset1, subset2 = datasets.split_dataset(original, 2, [2, 0, 3, 1, 4])
+        self.assertEqual(len(subset1), 2)
+        self.assertEqual(subset1[0], 3)
+        self.assertEqual(subset1[1], 1)
+        self.assertEqual(len(subset2), 3)
+        self.assertEqual(subset2[0], 4)
+        self.assertEqual(subset2[1], 2)
+        self.assertEqual(subset2[2], 5)
+
```

Comment:
Please add test cases where the length of permutation array is different from that of the underlying array.